charms/reactive/test.py

```diff
@@ -0,0 +1,69 @@
+import mock
+from unittest import TestCase
+
+
+class ReactiveTestCase(TestCase):
+    @classmethod
+    def _mock(cls, target, repl=mock.sentinel.DEFAULT):
+        if not hasattr(cls, '_cleanups'):
+            cls._cleanups = []
+        patcher = mock.patch(target, repl)
+        cls._cleanups.append(patcher.stop)
+        return patcher.start()
+
+    @classmethod
+    def setUpClass(cls):
+        cls._mock('charmhelpers.core.unitdata.kv').side_effect = NotImplementedError('Unmocked access to unitdata')
+        cls._mock('charms.reactive.relations.Conversation.join').return_value = cls
+        cls._mock('charms.reactive.relations.RelationBase.conversation').return_value = cls
+
+    @classmethod
+    def tearDownClass(cls):
+        for cleanup in cls._cleanups:
+            cleanup()
```

Comment:
https://docs.python.org/2/library/unittest.html#unittest.TestCase.doCleanups 
so shouldn't this use the builting addCleanup?